### PR TITLE
Ecommerce

### PR DESCRIPTION
**Pull request by pbright**
Matt asked for this to always be included.

I've assumed the existence of an analytics block in the base content.

### CODE REVIEW

**Review by gregplaysguitar — COMMENTED** [review]
Does this need to save a flag to the database? Since it sends google an id (order.invoice_number) that bit seems unnecessary?

File: checkout/jinja2/checkout/success.html [hunk 1]
```diff
@@ -3,6 +3,34 @@
...
 {% block title %}Order Completed{% endblock %}
 
+{% block analytics %}
+  {{ super() }}
+  {% if ecommerce %}
+    <script type="text/javascript">
+      console.log(window.ga);
+      if (typeof window.ga !== 'undefined' && window.ga !== null) {
+        ga('require', 'ecommerce');
+        ga('ecommerce:addTransaction', {
```
Comment (gregplaysguitar) [inline checkout/jinja2/checkout/success.html]: Should we add a `ga('ecommerce:clear');` here?

**Comment by pbright** [comment]
It is necessary, see http://stackoverflow.com/questions/26697898/google-analytics-duplicate-transactions.

File: checkout/jinja2/checkout/success.html (hunk 1 above)
Comment (pbright) [reply checkout/jinja2/checkout/success.html]: Good idea, I've added it.

**Comment by gregplaysguitar** [comment]
Ah, ok. I'm pretty keen not to pollute the database with flags for every tracking system - I think the best way to handle this would be to have a thanks page which the user only sees once, separate from the order detail page they can go back to. That'd mean changing the view structure a bit though. 

In the mean time can we just rename the flag so it's general, "thanks_page_viewed" or something? Then we can use it for other systems too with no further additions

**Comment by gregplaysguitar** [comment]
or "tracking_displayed" ?

**Comment by pbright** [comment]
Done, I like tracking_displayed better.

**Pull request merged**